### ISSUE TITLE
Very noisy training curves when trying to train with paired data, what am I doing wrong?

### ISSUE THREAD

**Issue by roey1rg** [xinntao/Real-ESRGAN#294]
I'm trying to train Real-ESRGAN with 400 images. 
The training curves are very noisy, especially the perceptual loss (l_g_precep).
I've tried to use various different learning rates.
I'm using 4 GPUs with 5 images on each GPU
![image](https://user-images.githubusercontent.com/44729747/162608284-e265ec55-d0d6-4476-8351-407b7243037a.png)

here is the config file I use:

# general settings
name: T8 # I changed 
model_type: RealESRGANModel
scale: 2  # I changed
num_gpu: auto
manual_seed: 0

# USM the ground-truth
l1_gt_usm: True
percep_gt_usm: True
gan_gt_usm: False

high_order_degradation: False # do not use the high-order degradation generation process

# dataset and data loader settings
datasets:
  train:
    name: sisr_test  # I changed
    type: RealESRGANPairedDataset
    dataroot_gt: /superrendering_data  # I changed
    dataroot_lq: /superrendering_data  # I changed
    meta_info: /superrendering_data/meta_info.txt  # I changed
    io_backend:
      type: disk

    gt_size: 256
    use_hflip: True
    use_rot: False

    # data loader
    use_shuffle: true
    num_worker_per_gpu: 3
    batch_size_per_gpu: 4
    dataset_enlarge_ratio: 1
    prefetch_mode: ~

  # Uncomment these for validation
  # val:
  #   name: validation
  #   type: PairedImageDataset
  #   dataroot_gt: path_to_gt
  #   dataroot_lq: path_to_lq
  #   io_backend:
  #     type: disk

# network structures
network_g:
  type: RRDBNet
  num_in_ch: 3
  num_out_ch: 3
  num_feat: 64
  num_block: 23
  num_grow_ch: 32
  scale: 2

network_d:
  type: UNetDiscriminatorSN
  num_in_ch: 3
  num_feat: 64
  skip_connection: True

# path
path:
  # use the pre-trained Real-ESRNet model
  pretrain_network_g: experiments/pretrained_models/RealESRGAN_x2plus.pth
  param_key_g: params_ema
  strict_load_g: true
  pretrain_network_d: experiments/pretrained_models/RealESRGAN_x2plus_netD.pth
  param_key_d: params
  strict_load_d: true
  resume_state: ~

# training settings
train:
  ema_decay: 0.999
  optim_g:
    type: Adam
    lr: !!float 1e-5  # I changed
    weight_decay: 0
    betas: [0.9, 0.99]
  optim_d:
    type: Adam
    lr: !!float 1e-4  # I changed
    weight_decay: 0
    betas: [0.9, 0.99]

  scheduler:
    type: MultiStepLR
    milestones: [200000, 400000]
    gamma: 0.5

  total_iter: 400000
  warmup_iter: -1  # no warm up

  # losses
  pixel_opt:
    type: L1Loss
    loss_weight: 1.0
    reduction: mean
  # perceptual loss (content and style losses)
  perceptual_opt:
    type: PerceptualLoss
    layer_weights:
      # before relu
      'conv1_2': 0.1
      'conv2_2': 0.1
      'conv3_4': 1
      'conv4_4': 1
      'conv5_4': 1
    vgg_type: vgg19
    use_input_norm: true
    perceptual_weight: !!float 1.0
    style_weight: 0
    range_norm: false
    criterion: l1
  # gan loss
  gan_opt:
    type: GANLoss
    gan_type: vanilla
    real_label_val: 1.0
    fake_label_val: 0.0
    loss_weight: !!float 1e-1 # !!float 1e-1  # I changed

  net_d_iters: 1
  net_d_init_iters: 0

# Uncomment these for validation
# validation settings
# val:
#   val_freq: !!float 5e3
#   save_img: True

#   metrics:
#     psnr: # metric name
#       type: calculate_psnr
#       crop_border: 4
#       test_y_channel: false

# logging settings
logger:
  print_freq: 20  # I changed from 100
  save_checkpoint_freq: !!float 5e3
  use_tb_logger: true
  wandb:
    project: ~
    resume_id: ~

# dist training settings
dist_params:
  backend: nccl
  port: 29500

**Comment by JensDA** [comment]
The logged losses are only averages of the mini batch of the one iteration at the time of logging.
They are not averaged over the iterations between logs, so they are more like regular samples and thus are prone to noise.

**Comment by roey1rg** [comment]
Thank @JensDA, good point

**Comment by mindFlayer123** [comment]
@roey1rg where to find the log file ?

**Comment by Tsouuu** [comment]
> I'm trying to train Real-ESRGAN with 400 images. The training curves are very noisy, especially the perceptual loss (l_g_precep). I've tried to use various different learning rates. I'm using 4 GPUs with 5 images on each GPU ![image](https://user-images.githubusercontent.com/44729747/162608284-e265ec55-d0d6-4476-8351-407b7243037a.png)
> 
> here is the config file I use:
> 
> ![output-onlineyamltools (1)](https://user-images.githubusercontent.com/44729747/162608495-c8ace47c-0636-455c-9a53-321b7e7c8055.png)
Can you explain how to obtain that plot? plz

**Comment by FengMu1995** [comment]
> I'm trying to train Real-ESRGAN with 400 images. The training curves are very noisy, especially the perceptual loss (l_g_precep). I've tried to use various different learning rates. I'm using 4 GPUs with 5 images on each GPU ![image](https://user-images.githubusercontent.com/44729747/162608284-e265ec55-d0d6-4476-8351-407b7243037a.png)
> 
> here is the config file I use:
> 
> ![output-onlineyamltools (1)](https://user-images.githubusercontent.com/44729747/162608495-c8ace47c-0636-455c-9a53-321b7e7c8055.png)

My l_g_percep loss have been increasing gradually, do you know why it haven't decreased?
